src/lib/configs/project/dockerCompose.ts

```diff
@@ -329,3 +401,162 @@ export const initReadonlyDockerComposeConfig =
   getReadonlyConfigInitFunction(initConfigOptions);
 
 export const dockerComposeSchema: JSONSchemaType<LatestConfig> = configSchemaV0;
+
+export async function ensureConfigToml(
+  fluenceConfig: FluenceConfigReadonly | null,
+  providerConfig: ProviderConfigReadonly,
+) {
+  const baseNoxConfig = mergeNoxConfigYAML(
+    await getDefaultNoxConfigYAML(fluenceConfig),
+    providerConfig.nox ?? {},
+  );
+
+  const configsDir = await ensureFluenceConfigsDir();
+  const { stringify } = await import("@iarna/toml");
+
+  await Promise.all(
+    Object.entries(providerConfig.computePeers).map(async ([key, value], i) => {
+      const overridden = mergeNoxConfigYAML(baseNoxConfig, value.nox ?? {});
+
+      if (overridden.tcpPort === TCP_PORT_START) {
+        overridden.tcpPort = TCP_PORT_START + i;
+      }
+
+      if (overridden.websocketPort === WEB_SOCKET_PORT_START) {
+        overridden.websocketPort = WEB_SOCKET_PORT_START + i;
+      }
+
+      if (overridden.httpPort === HTTP_PORT_START) {
+        overridden.httpPort = HTTP_PORT_START + i;
+      }
+
+      return writeFile(
+        join(configsDir, getConfigTomlName(key)),
+        [
+          stringify(configYAMLToConfigToml(overridden)),
+          providerConfig.nox?.rawConfig ?? "",
+          value.nox?.rawConfig ?? "",
+        ]
+          .filter((x) => {
+            return x !== "";
+          })
+          .join("\n"),
+        FS_OPTIONS,
+      );
+    }),
+  );
+}
+
+function mergeNoxConfigYAML(a: NoxConfigYAML, b: NoxConfigYAML): NoxConfigYAML {
+  return {
+    ...a,
+    ...b,
+    ...(a.systemServices !== undefined && b.systemServices !== undefined
+      ? {
+          systemServices: {
+            ...a.systemServices,
+            ...b.systemServices,
+            enable: Array.from(
+              new Set([
+                ...(a.systemServices.enable ?? []),
+                ...(b.systemServices.enable ?? []),
+              ]),
+            ),
+            aquaIpfs: {
+              ...a.systemServices.aquaIpfs,
+              ...b.systemServices.aquaIpfs,
+            },
+            decider: {
+              ...a.systemServices.decider,
+              ...b.systemServices.decider,
+            },
+          },
+        }
+      : {}),
+  };
+}
```

Comment:
OMG! Can we use something like `lodash.merge` ?

Reply:
I first used merge from lodash and find out it doesn't merge the way I want it to, but I checked out and there is a mergeWith which I can use so Arrays are overrided instead of weirdly merged (by default in arrays items are overriden by index so if you have `prop:  ['a', 'b']` in one config and `prop: ['c']` in another you will get `prop: ['c', 'b']`. I think it's better to just get `prop: ['c']` in the end result because we want users to be able to override configs completely)